src/tools/stardb/startextdump.cpp

```diff
@@ -250,9 +250,30 @@ bool DumpStarDatabase(istream& in, ostream& out)
 
         out << catalogNum << ' ';
         out << setprecision(7);
-        out << x << ' ' << y << ' ' << z << ' ';
-        out << setprecision(4);
-        out << ((float) absMag / 256.0f) << ' ';
+        
+        if (spherical)
+        {
+            Eigen::Vector3d eclipticpos = {(double) x, (double) y, (double) z};
+            Eigen::Vector3d pos = astro::eclipticToEquatorial(eclipticpos);
+            double distance = sqrt(x * x + y * y + z * z);
+            double phi = -acos(pos.y() / distance) * 180 / PI;
+            double theta = atan2(pos.z(), -pos.x()) * 180 / PI;
+            double ra = theta + 180;
```

Comment:
if `θ = RA + 180°` then `RA = θ - 180`

Reply:
I used `RA = θ + 180` instead of `RA = θ - 180` because atan2 outputs angles in the interval [-π, π], not [0, 2π] like you (and I) might expect. To me it makes much more sense just to add 180 degrees instead of subtract 180 degrees, even though they're equivalent.

If you want though, I could make this more clear by doing `ra = theta - 180 + 360;`

Reply:
yes, it's better to use `- 180 + 360` and leave a comment "atan2 return result in [-pi; pi] instead of [0, 2pi], so we add 360 to fix this"